src/openms/source/SIMULATION/MSSim.cpp

```diff
@@ -299,6 +299,7 @@ namespace OpenMS
       SignedSize scan_index = distance<SimTypes::MSSimExperiment::ConstIterator>(experiment_.begin(), it_rt);
       pi.setMetaValue("RT_index", scan_index);
       pi.setRT(f.getRT());
+      pi.setMZ(f.getMZ());
```

Comment:
I guess this was written when mz was still a metavalue.
Anyways:
You could try: `if (!hasMZ()) setMZ(...))`. If the m/z was jittered before (to simulate precision losses), it would stay like that.